### PR TITLE
Share button

### PR DESCRIPTION
**Pull request by me-liu**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- share button opens a tab from the bottom, shares a link to the Add Case tab 
- there is an image

### Screenshots

 
IMG_4869

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[Deep Linking Base](https://github.com/calblueprint/impact-fund/pull/74)

CC: @ronniebeggs

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,7 +53,8 @@
     "uri-scheme": "^1.2.0",
     "yargs": "^16.2.0",
     "yargs-parser": "^20.2.9",
-    "zod": "^3.22.4"
+    "zod": "^3.22.4",
+    "expo-sharing": "~11.5.0"
```
Comment (ronniebeggs) [inline package.json]: Remove `expo-sharing` from the dependencies if we don't end up needing it.

File: src/app/(BottomTabNavigation)/AllCases/CaseScreen/[caseUid].tsx [hunk 2]
```diff
@@ -25,6 +27,21 @@ function CaseScreen() {
     setIsLoading(false);
   };
 
+  const onShare = async () => {
+    const addCaseUrl = Linking.createURL('?event=addCase&', {
+      queryParams: { caseUid },
+    });
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/CaseScreen/[caseUid].tsx]: `event=addCase` is technically a query parameter, so maybe we can clean up this function instead of concatenating multiple partial strings.

File: src/app/(BottomTabNavigation)/AllCases/CaseScreen/[caseUid].tsx [hunk 3]
```diff
@@ -25,6 +27,21 @@ function CaseScreen() {
...
+    console.log(addCaseUrl);
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/CaseScreen/[caseUid].tsx]: Remove log statement before merge.

File: src/app/(BottomTabNavigation)/AllCases/CaseScreen/styles.ts [hunk 4]
```diff
@@ -38,4 +37,17 @@ export default StyleSheet.create({
     fontWeight: 'bold',
     textAlign: 'center',
   },
+  shareContainer: {
+    justifyContent: 'flex-end',
+    flexDirection: 'row',
+    marginTop: 10,
+    marginLeft: 280,
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/CaseScreen/styles.ts:45]: Try putting the share container in a larger container and pushing it to the right using `flex-end`. Or, because the main content occupies `92%` of the screen width, I'd even prefer a `marginRight: '4%'` so it adjusts according to screen size.

File: src/app/(BottomTabNavigation)/AllCases/CaseScreen/[caseUid].tsx [hunk 5]
```diff
@@ -25,6 +27,21 @@ function CaseScreen() {
...
+    try {
+      await Share.share({
+        //message: 'ADD CASE',
+        url: `${addCaseUrl}/`,
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/CaseScreen/[caseUid].tsx]: Out of curiosity, is there anyway to customize how the link looks when shared with someone? Sometimes apps will attach an image or additional text to the link so users don't have to see the entire link.

**Review by ronniebeggs — CHANGES_REQUESTED** [review]
Looks good so far! I left a few questions and comments.
